docs: reboot CONTRIBUTING.md and PR template to reflect 4-stage manifesto (#18708)
LedgerHQ/ledger-live: CONTRIBUTING.md

    S2 --> S3[Pass review<br>by code-owners]
```

### Automated checks

Before marking your PR ready for review, ensure all of the following pass:

- **CI linters** — lint and TypeScript checks must pass.
- **CI tests** — unit tests and e2e tests must be green.
- **SonarQube** — address all findings: fix, mark as accepted debt, or flag as false positive. See [`docs/contributing/sonarqube-guide.md`](docs/contributing/sonarqube-guide.md).
- **Copilot** — request a Copilot review as early as Draft, address or explicitly dismiss every comment, and resolve all threads.
  <img width="353" height="164" alt="Request Copilot on a Draft PR" src="https://github.com/user-attachments/assets/1326c947-61bc-4793-b70c-9e39b04eb630" />
  If you need to re-trigger Copilot after pushing new commits, re-request it the same way.
  <img width="361" height="104" alt="Re-request review of Copilot to retrigger it" src="https://github.com/user-attachments/assets/8109025c-ba89-4f17-8496-47bacc694cca" />

For testing requirements see the [Developer Portal](https://developers.ledger.com/docs/ledger-live/contributing/reference/testing).

### Code owner review

> 

Reviewer: **suggestion: keep focus of CONTRIBUTING.md to PR authors**

When reading down the file the imagined audience of the file suddenly changes at this point, from the author to the reviewer. 

I can see two ways to go:

1. Re-write this from the perspective of the author, e.g. "If you see an unexpected reviewers on your PR it may be appropriate to update the CODEOWNERS file"
2. Move the guidelines for the reviewer out of this file.

I think both ways are worth following. i.e. move the guidelines for reviewers into `REVIEWERS.md` (currently `docs/contributing/pr-review-guide.md`) and also considering whether to maintain an equivent guideline for the author, as per the example above.

# Contributing to Ledger Live

> Thanks for contributing. Please read this guide fully before opening a pull request — it describes the workflow the team has converged on for 2025/2026, including the AI-assisted stages that gate every PR.

Reviewer: **suggestion: assert that these guidelines apply to internal contributors as well**
**plus: keep it timeless**

```suggestion
Thanks for contributing! These guidelines are for internal and external contributors, including agents. Please read fully before opening a pull request — it describes the workflow for contributing, including AI-assisted stages that gate every PR.
```

## Table of Contents

1. [Disclaimer](#disclaimer)
2. [Getting Started](#getting-started)
3. [Branch & Commit Conventions](#branch--commit-conventions)
4. [The PR Lifecycle](#the-pr-lifecycle)
5. [Review Culture](#review-culture)
6. [Changelogs](#changelogs)
7. [Translations](#translations)
8. [Developer Portal](#developer-portal)

## Disclaimer

> For **Ledger Live Desktop and Mobile** we are currently accepting bug fixes and invited contributions only. Feature PRs that do not align with our roadmap or long-term goals will be closed without extensive review.

Reviewer: **suggestion: offer a channel for feature requests**

IMO we should direct people who want to contribute features towards some course of action. For example: describing their idea on Discord, opening a GitHub issue or contacting Ledger support. 

WDYT?

Reviewer: On reflection, I recognise it's outside the scope of the PR but it would still be good to get your thoughts

Reviewer 2: (after discussion, we will not touch this from now, this got inherited from previous CONTRIBUTING.md state, we will have to internal discuss in future iteration)

> For **Ledger Live Desktop and Mobile** we are currently accepting bug fixes and invited contributions only. Feature PRs that do not align with our roadmap or long-term goals will be closed without extensive review.

Reviewer: **suggestion: let's use "Ledger Wallet" rather than "Ledger Live"**

This was discussed elsewhere (maybe on another PR). The [main README](https://github.com/LedgerHQ/ledger-live/blob/develop/README.md) now mentions the rename so everywhere else I think we should move to "Ledger Wallet"

```suggestion
For **Ledger Wallet** is currently accepting bug fixes and invited contributions only. Feature PRs that do not align with our roadmap or long-term goals will be closed without extensive review.
```

Reviewer: **Updated suggestion**

The previous wording was more flexible. WDYT to this?

```suggestion
**Ledger Wallet** is currently only accepting bug fixes and invited contributions. There is a good chance that feature PRs that do not align with our roadmap will be closed without extensive review.
```

Reviewer 2: let's go back to previous wording but just fix the "Live" to "Wallet".

## Getting Started

> 1. Fork the repository and create your branch from `develop` (see [Branch naming](#branch-naming) below).

Reviewer: **suggestion: stay internal-external agnostic**

```suggestion
0. External contributors should fork the repository.
1. Create your branch from `develop` (see [Branch naming](#branch-naming) below).
```

> 2. Follow the main installation guide: <https://github.com/LedgerHQ/ledger-live#installation>

Reviewer: **request: correct this broken link**

```suggestion
2. Follow the main README for getting started: https://github.com/LedgerHQ/ledger-live#getting-started
```

> 3. Follow any additional setup steps for the specific package you are working on.

Reviewer: **suggestion: clarify process for app and lib development**

```suggestion
3. Follow additional setup instructions in the README files of the app or lib you are working on.
```

## Branch & Commit Conventions

### Branch naming

| Prefix | When to use |
|--------|-------------|
| `feat/` | Adding a new feature |
| `bugfix/` | Fixing a bug |
| `support/` | Refactors, tests, CI, tooling improvements |

### Commit messages

We follow [Conventional Commits](https://www.conventionalcommits.org/) and enforce it with [commitlint](https://commitlint.js.org/).

Use `pnpm commit` for an interactive prompt, or `pnpm commitlint --from <target-branch>` to validate your branch.

### Rebase & merge strategy

**Always prefer rebasing** unless your branch contains merge commits from sub-features.

- Small, self-contained branches → rebase on `develop`.
- Branches with cross-branch merges → merge `develop` into them to stay up to date.

> > ⚠️ Do **not** rebase a branch that is awaiting translations from Smartling.

Reviewer: **query: is the rebase option available to all contributors?**

Do all contributors have the option to merge or rebase? Your GitHub privileges may be different. I think I only have the option of a merge commit. We can check together if you like.

```suggestion
```

Reviewer 2: yes anyone can Rebase a branch. what the doc suggest here is to keep a PR branch maintained and rebased (not contain merge commits). regardless of the "Merge button" part. 🤔  does it make sense?

btw this is inherited from https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#rebase--merge-strategies but was just revamped

Reviewer: Ah I see. Perhaps change the heading to something like, "Updating your branch"

When I read, "Rebase [...] merge strategy" I was thinking about merging the PR rather than updating the branch.

We have to be careful this the instruction to "update the branch" as well. We all know that it could be a loosing battle.

How about something like this: -

### Updating your branch

Keep your branch up to date with `develop`, at least every 24 hours.

- **Don't branch off a branch** – keep your branch small, self-contained and based on `develop`
- **Prefer rebasing** – stay up to date with `develop` by "rebasing" rather than "merging"
- **Avoid rebasing when...** your branch contains merge commits or is awaiting translations from Smartling

> ## The PR Lifecycle

Reviewer: **request: can we make the "PR Lifecycle" section snappier?**

A lot of what is under this heading is overly verbose, prescriptive and opinionated. e.g. "Fix any failing checks before moving on. Do not ask others to review a red PR" and it would be better to stay high level. 

e.g. 

- **Ensure all CI checks pass** - see `github/workflows` files or GitHub UI for full details.
- **Address comments from automated reviewers** – Review from Copilot can be requested.  
- **Pass SonarQube checks** – see [sonarqube-guide](./docs/contributing/sonarqube-guide.md) if SonarQube is unhappy with your changes.

---

Keeping this high level has the additional advantage that low-level details can change without this guide needing to be updated.

> Every PR goes through four stages. **Open your PR as a Draft** and progress through the stages before requesting human review.

Reviewer: **suggestion: use this line to give a high level vision of the process:**

```suggestion
Open your PR as a **Draft** and pass all automated-checks before making **Ready for Review**:
```

```mermaid
flowchart LR
    Draft --> S0[Stage 0\nCI passes]
    S0 --> S1[Stage 1\nSonarQube]
    S1 --> S2[Stage 2\nCopilot review]
    S2 --> Opened

>     Opened --> S3[Stage 3\nCode owner review]

Reviewer: **suggestion: let's mirror the language of GitHub UI**

GitHub is a bit ambiguous with their language here regarding PRs. They do list "Draft" in a list called "Open". 

The term "Ready for review" is less ambiguous. It used on the CTA of a draft PR. 

```suggestion
    Draft --> S0[Stage 0\nCI passes]
    S0 --> S1[Stage 1\nSonarQube]
    S1 --> S2[Stage 2\nCopilot review]
    S2 --> Open
    Open --> S3[Stage 3\nCode owner review]
```

or 

```suggestion
    Draft --> S0[Stage 0\nCI passes]
    S0 --> S1[Stage 1\nSonarQube]
    S1 --> S2[Stage 2\nCopilot review]
    S2 --> Ready for review
    Ready for review --> S3[Stage 3\nCode owner review]
```

Reviewer: Closing this in favour of a higher level suggestion

> ```

Reviewer: **suggestion: let's simplify this workflow**

```mermaid
flowchart LR
    S0[Create draft\npull request] --> S1[Pass all\nautomated checks]
    S1 --> S2[Open pull request:\nReady for review]
    S2 --> S3[Pass review\nby code-owners]
```

Reviewer 2: agreed. and then we can detail in short the automated checks to expect. I would like to make it clear that SonarQube ✅  and global GH PR green status is expected but we don't have to go too much in depth. maybe just a basic bullet point concise list. (sonarqube is one bullet point and we can link to the SonarQube guide)

> ### Stage 0 — CI passes

Reviewer: Let's merge all the automated check into a single stage, including SonarQube and GitHub Copilot's automated-reviewer?

- Use this time to let AI agents (Copilot, your local assistant) iterate on the code.

### Stage 1 — SonarQube

SonarQube runs automatically on every PR and analyses code quality, security hotspots, and test coverage.

- Address all SonarQube findings before proceeding: fix them, mark them as accepted technical debt with an explanation, or flag them as false positives.
- Do not silently ignore findings — unaddressed issues block the review.
- Aim for meaningful test coverage on new logic. 0 % is not acceptable.

See [`docs/contributing/sonarqube-guide.md`](docs/contributing/sonarqube-guide.md) for a walkthrough of the SonarQube UI and how to handle each finding type.

### Stage 2 — Copilot review (AI gate)

- Manually request a **Copilot** review from the Reviewers panel.
- Read every Copilot comment and either address it or leave an explicit reply explaining why it does not apply.
- All Copilot threads must be resolved before you proceed.

> ✅ After completing stage 2, convert the PR from **Draft → Open**. This is the signal to code owners that the PR is ready for human review.

### Stage 3 — Code owner review

> - Opening the PR (non-draft) automatically requests the relevant code owners via `CODEOWNERS`.

Reviewer: I think we're describing too much detail here.  Let's keep the focus on the requirements and if we want to go into the details of our decision it should be in an ADR.

The more consistently everyone does this, the less individual pinging is needed, and the shorter review cycles become.

## Review Culture

- **Review the files you own.** Use GitHub's "Files changed" → "Viewed" filter to focus on files relevant to your team.
- **A PR is only as ready as its worst stage.** If CI is red or Copilot feedback is unaddressed, it is not ready — move it back to Draft.
- **Resolve before merging.** All review comments should be resolved or explicitly acknowledged before merge.
- **PRs must have value.** Consider whether the change is worth the review cost before opening it. Close low-value or abandoned PRs rather than letting them accumulate.

## Changelogs

We use [changesets](https://github.com/changesets/changesets) for versioning. Run:

```bash
pnpm changeset
```

A changeset is **required** for any user-facing change or library API modification. See the [wiki](https://github.com/LedgerHQ/ledger-live/wiki/Changesets) for the full guide.

## Translations

We use [Smartling](https://www.smartling.com/) for automated translations.

> > ⚠️ **Only edit the English source files.** Never commit files for other locales.

Reviewer: **suggestion: move the translations details to the specific app READMEs**

This duplicates parts of README files in LWD and LWM. 

If there are details here that we need to preserve, let's move them to those READMEs. I don't think it's globally relevant context. WDYT?

> # Contributing to Ledger Live

Reviewer: **suggestion: let's go with Ledger Wallet**

I think I commented about this already somewhere. Happy to elaborate if necessary

```suggestion
# Contributing to Ledger Wallet
```
